command/helper/helper.go

```diff
@@ -143,3 +171,310 @@ func GenerateUsage(baseCommand string, flagMap map[string]FlagDescriptor) string
+				"dev": map[string]interface{}{},
+			},
+		},
+		Bootnodes: []string{},
+	}
+
+	if err := FillPremineMap(cc.Genesis.Alloc, premine); err != nil {
+		return err
+	}
+
+	return WriteGenesisToDisk(cc, genesisPath)
+}
+
+// BootstrapDevCommand creates a config and generates the dev genesis file
+func BootstrapDevCommand(baseCommand string, args []string) (*Config, error) {
+	config := DefaultConfig()
+
+	cliConfig := &Config{
+		Network: &Network{
+			NoDiscover: true,
+			MaxPeers:   0,
+		},
+	}
+	cliConfig.Seal = true
+	cliConfig.Dev = true
+	cliConfig.Chain = "genesis.json"
+
+	flags := flag.NewFlagSet(baseCommand, flag.ContinueOnError)
+	flags.Usage = func() {}
+
+	var premine helperFlags.ArrayFlags
+
+	flags.StringVar(&cliConfig.LogLevel, "log-level", DefaultConfig().LogLevel, "")
+	flags.Var(&premine, "premine", "")
+	flags.Uint64Var(&cliConfig.DevInterval, "dev-interval", 0, "")
+
+	if err := flags.Parse(args); err != nil {
+		return nil, err
+	}
+
+	if err := config.mergeConfigWith(cliConfig); err != nil {
+		return nil, err
+	}
+
+	if err := generateDevGenesis(config.Chain, premine); err != nil {
+		return nil, err
+	}
+
+	return config, nil
+}
+
+func ReadConfig(baseCommand string, args []string) (*Config, error) {
+	config := DefaultConfig()
+
+	cliConfig := &Config{
+		Network: &Network{},
+	}
+
+	flags := flag.NewFlagSet(baseCommand, flag.ContinueOnError)
+	flags.Usage = func() {}
+
+	var configFile string
+	flags.StringVar(&cliConfig.LogLevel, "log-level", "", "")
+	flags.BoolVar(&cliConfig.Seal, "seal", false, "")
+	flags.StringVar(&configFile, "config", "", "")
+	flags.StringVar(&cliConfig.Chain, "chain", "", "")
+	flags.StringVar(&cliConfig.DataDir, "data-dir", "", "")
+	flags.StringVar(&cliConfig.GRPCAddr, "grpc", "", "")
+	flags.StringVar(&cliConfig.JSONRPCAddr, "jsonrpc", "", "")
+	flags.StringVar(&cliConfig.Join, "join", "", "")
+	flags.StringVar(&cliConfig.Network.Addr, "libp2p", "", "")
+	flags.StringVar(&cliConfig.Network.NatAddr, "nat", "", "the external IP address without port, as can be seen by peers")
+	flags.BoolVar(&cliConfig.Network.NoDiscover, "no-discover", false, "")
+	flags.Uint64Var(&cliConfig.Network.MaxPeers, "max-peers", 0, "")
+	flags.BoolVar(&cliConfig.Dev, "dev", false, "")
+	flags.Uint64Var(&cliConfig.DevInterval, "dev-interval", 0, "")
+
+	if err := flags.Parse(args); err != nil {
+		return nil, err
+	}
+
+	if configFile != "" {
+		// A config file has been passed in, parse it
+		diskConfigFile, err := readConfigFile(configFile)
+		if err != nil {
+			return nil, err
+		}
+
+		if err := config.mergeConfigWith(diskConfigFile); err != nil {
+			return nil, err
+		}
+	}
+
+	if err := config.mergeConfigWith(cliConfig); err != nil {
+		return nil, err
+	}
+
+	return config, nil
+}
+
+type HelpGenerator interface {
+	DefineFlags()
+}
+
+// Meta is a helper utility for the commands
+type Meta struct {
+	UI   cli.Ui
+	Addr string
+
+	FlagMap        map[string]FlagDescriptor
+	HasGlobalFlags bool
```

Comment:
HasGlobalFlags is only assigned to true in the below method and that's it. We can remove it.

Reply:
I agree. It was part of an earlier implementation that required a different structure for `Meta`, but it's useless now 👍